src/rendering/SceneManager.cc

```diff
@@ -925,6 +926,76 @@ rendering::VisualPtr SceneManager::CreateActor(Entity _id,
   return actorVisual;
 }
 
+/////////////////////////////////////////////////
+rendering::VisualPtr SceneManager::CreateLightVisual(Entity _id,
+    const sdf::Light &_light, Entity _parentId)
+{
+  if (!this->dataPtr->scene)
+    return rendering::VisualPtr();
+
+  if (this->dataPtr->visuals.find(_id) != this->dataPtr->visuals.end())
+  {
+    ignerr << "Entity with Id: [" << _id << "] already exists in the scene"
+           << std::endl;
+    return rendering::VisualPtr();
+  }
+
+  std::string name = _light.Name().empty() ? std::to_string(_id) :
+      _light.Name();
+
+  rendering::LightPtr lightParent;
+  auto it = this->dataPtr->lights.find(_parentId);
+  if (it != this->dataPtr->lights.end())
+  {
+    lightParent = it->second;
+  }
+  else
+  {
+    ignerr << "Parent entity with Id: [" << _parentId << "] not found. "
+           << "Not adding light visual with ID[" << _id
+           << "]  and name [" << name << "] to the rendering scene."
+           << std::endl;
+    return rendering::VisualPtr();
+  }
+
+  name = lightParent->Name() +  "::" + name + "Visual";
+
+  if (this->dataPtr->scene->HasVisualName(name))
+  {
+    ignerr << "Visual: [" << name << "] already exists" << std::endl;
```

Comment:
I'm hitting this line when I do:

1. Insert a spot light
2. Delete the spot light
3. Insert a new spot light, and I see:
    `[GUI] [Err] [SceneManager.cc:965] Visual: [spotlight::spotlightVisual] already exists`

I believe the light visual isn't being deleted with the light, and this can cause large memory leaks down the road. We should make sure the light visual is deleted together with the light.